parameter/ConfigurableDomain.cpp

```diff
@@ -769,7 +769,10 @@ bool CConfigurableDomain::getElementSequence(const string& strConfiguration, str
     return true;
 }
 
-bool CConfigurableDomain::setApplicationRule(const string& strConfiguration, const string& strApplicationRule, const CSelectionCriteriaDefinition* pSelectionCriteriaDefinition, string& strError)
+bool CConfigurableDomain::setApplicationRule(const string& strConfiguration,
+                                             const string& strApplicationRule,
+                                             const core::selection::criterion::Criteria& criteria,
```

Comment:
you're right, I used it for consistency. 
In fact I thought it could be useful to later add rules to the selection namespace but for now YAGNI.